### PR TITLE
CreateVacancyAd endpoint

### PR DESCRIPTION
**Pull request by blazejb**

### CODE REVIEW

File: lambdas/vacancy-ad/create-vacancy-ad.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    .executeTakeFirstOrThrow();
+
+  const vacancyAdCategoryValues = input.categoryIds.map((id) => ({
+    vacancyAdId: sql`uuid(${vacancyAdResult.id})`.castTo<string>(),
+    categoryId: sql`uuid(${id})`.castTo<string>(),
+  }));
+
+  await trx.insertInto('vacancyAdCategory').values(vacancyAdCategoryValues).execute();
+
+  const vacancyAd = await trx
```
Comment (tbekas) [inline lambdas/vacancy-ad/create-vacancy-ad.ts:78]: You don't need to fetch that again, you already have it in `vacancyAdResult`

File: lambdas/vacancy-ad/vacancy-ad-mapping.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import { VacancyAdRow } from '../database/schema/vacancy-ad-table';
+
+export function toVacancyAdDto({ id, name, description, collaborative, active, createdAt }: VacancyAdRow): VacancyAd {
+  return {
+    id,
+    name,
+    description,
+    collaborative,
+    active,
+    createdAt: toAWSDateTime(createdAt),
```
Comment (tbekas) [inline lambdas/vacancy-ad/vacancy-ad-mapping.ts:12]: The `categories` field will be `null` even though `schema.graphql` says it shouldn't be.

**Review by tbekas — CHANGES_REQUESTED** [review]
Overall it looks very nice and clean!

File: lambdas/vacancy-ad/vacancy-ad-mapping.ts (hunk 2 above)
Comment (blazejb) [reply lambdas/vacancy-ad/vacancy-ad-mapping.ts:12]: Removed those fields from` schema.graphql` for now. I will add them in `vacancyAd` fetching task.

Comment (tbekas) [reply lambdas/vacancy-ad/vacancy-ad-mapping.ts:12]: Ok 👍

**Review by tbekas — APPROVED** [review]

**Pull request merged**